src/esthri/src/ops/sync.rs

```diff
@@ -153,25 +129,25 @@ where
     Ok(())
 }
 
-fn process_globs<'a, P: AsRef<Path> + 'a>(
-    path: P,
-    glob_includes: &[Pattern],
-    glob_excludes: &[Pattern],
-) -> Option<P> {
+fn process_globs<'a, P: AsRef<Path> + 'a>(path: P, filters: &[GlobFilter]) -> Option<P> {
     let mut excluded = false;
     let mut included = false;
     {
         let path = path.as_ref();
-        for pattern in glob_excludes {
-            if pattern.matches(path.to_string_lossy().as_ref()) {
-                excluded = true;
-                break;
-            }
-        }
-        for pattern in glob_includes {
-            if pattern.matches(path.to_string_lossy().as_ref()) {
-                included = true;
-                break;
+        for pattern in filters {
+            match pattern {
+                GlobFilter::Include(filter) => {
+                    if filter.matches(path.to_string_lossy().as_ref()) {
+                        included = true;
+                        break;
```

Comment:
In order to implement the "left to right" processing that aws cli docs describe, don't we need to continue to process filters instead of breaking out?

Reply:
Hmm, I interpreted the docs a little differently:

> Any number of these parameters can be passed to a command. You can do this by providing an --exclude or --include argument multiple times, e.g. --include "*.txt" --include "*.png". When there are multiple filters, the rule is the filters that appear later in the command take precedence over filters that appear earlier in the command. For example, if the filter parameters passed to the command were
>
> `--exclude "*" --include "*.txt"`
>
> All files will be excluded from the command except for files ending with .txt However, if the order of the filter parameters was changed to
>
> `--include "*.txt" --exclude "*"`
>
> All files will be excluded from the command.

(from https://docs.aws.amazon.com/cli/latest/reference/s3/index.html#use-of-exclude-and-include-filters)

I read that to mean that it goes "right to left" and takes the first matching rule as the decision for the file. So, once you have a match then you can break out and say that's the decision. For the `aws s3` case, if there's no match then you fall through to the implicit `--include *` case and the file will be included.

Reply:
Your interpretation makes sense, but in that case, shouldn't we reverse the list of filters prior to processing them? (Or are we already)?

Reply:
Yeah, it's not super obvious but the list of filters is sorted in order from last specified to first when it's read from the CLI:

https://github.com/swift-nav/esthri/blob/c68a917bf8b33a2cf9646bb6396c716b5da5f91a/src/esthri/src/main.rs#L340-L375

IMO having the API accept the filters in the order that they'll be processed makes more sense than having it process the list in reverse order, so I reversed the order there.

Reply:
Ok, I was struggling to figure out where the reverse was happening, finally found it here: `filters.sort_by(|a, b| b.0.cmp(&a.0));` 😅

Can we try `filters.sort_by(|a, b| a.0.cmp(&b.0).reverse());` instead?

Reply:
Docs: https://doc.rust-lang.org/std/cmp/enum.Ordering.html#method.reverse

Reply:
Good call! done